File: 1486.xor-operation-in-an-array.js
/*
 * @lc app=leetcode id=1486 lang=javascript
 *
 * [1486] XOR Operation in an Array
 */

// @lc code=start
/**
 * @param {number} n
 * @param {number} start
 * @return {number}
 */
var xorOperation = function(n, start) {
   let result = 0
   for (let i=start; i<start+n*2; i+=2) {
     result ^= i
   }
   return result
};
// @lc code=end

File: 1486.xor-operation-in-an-array.py
#
# @lc app=leetcode id=1486 lang=python3
#
# [1486] XOR Operation in an Array
#

# @lc code=start
class Solution:
    def xorOperation(self, n: int, start: int) -> int:
        if n < 2: return start
        x = start
        for i in range(1, n):
            x = x ^ start+2*i

        return x
# @lc code=end

